src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -0,0 +1,116 @@
+import Boom from 'boom';
+import { get, omit } from 'lodash';
+
+import {
+  createFindQuery,
+  createFilterPath,
+  handleEsError,
+} from './lib';
+
+export class SavedObjectsClient {
+  constructor(kibanaIndex, callAdminCluster) {
+    this._kibanaIndex = kibanaIndex;
+    this._callAdminCluster = callAdminCluster;
+  }
+
+  async create(type, options = {}) {
+    const body = omit(options, 'id');
+
+    const response = await this._withKibanaIndex('index', {
+      type,
+      body
+    });
+
+    return Object.assign({
+      type: response._type,
+      id: response._id,
+      version: response._version
```

Comment:
Elasticsearch already set the precedent for using underscores as special fields, and that seems like a reasonable option to me.  We'd want to do that for all of our special fields though, so id and type as well.  We could outright deny setting any underscore-prefixed fields directly, as @spalger describes in his second point.

Everyone that touches Elasticsearch is used to prefixed field names for things like id and type, and that convention will certainly work for us, so let's not invent a new convention if we can avoid it.